4_visualize/src/plot_results.R

```diff
@@ -0,0 +1,412 @@
+#' ice, hice (ice thickness), evap (evaporation), n_layers (number of simulated 
+#' layers), depth_class (surface, middle, or bottom), depth, and temperature 
+get_surface_middle_bottom_preds <- function(run_groups) {
+  surface_middle_bottom_preds <- run_groups %>%
+    group_by(site_id) %>%
+    group_modify(~ {
+      site_meteo_files <- .x$raw_meteo_fl
+      site_export_files <- .x$export_fl
+      run_gcm_list <- .x$gcm
+      # Pull predictions for each site
+      site_preds <- get_site_preds(site_meteo_files, site_export_files, run_gcm_list)
+      
+      # Munge to long format
+      site_preds_long <- munge_long(site_preds)
+      
+      # Identify surface, middle, and bottom depths for each site and filter
+      # predictions to only those depths
+      plot_depths <- get_plotting_depths(site_preds_long)
+      filtered_preds <- filter(site_preds_long, depth %in% plot_depths) %>%
+        mutate(depth_class=case_when(
+          depth==plot_depths$plot_depth_surface ~ 'surface',
+          depth==plot_depths$plot_depth_middle ~ 'middle',
+          depth==plot_depths$plot_depth_bottom ~ 'bottom'
+        ), .before=depth)
+      return(filtered_preds)
+    })
+  return(surface_middle_bottom_preds)
+}
+
+#' @title Munge GLM predictions from wide to long format
+#' @description pivots wide-format GLM predictions (with columns like
+#' 'temp_0', 'temp_0.5', etc.) to wide format, with a column for 'depth'
+#' and a column for 'temperature'
+#' @param input_wide The wide-formatted input data
+#' @return a munged dataframe of GLM predictions that includes all of 
+#' the columns in the original wide data except the 'temp_X' columns,
+#' which are instead reduced to two columns: 'depth' and 'temperature'
+munge_long <- function(input_wide) {
+  input_long <- input_wide %>%
+    pivot_longer(temp_0:colnames(input_wide)[ncol(input_wide)], names_to="depth", values_to="temperature") %>%
+    mutate(depth = as.numeric(str_remove(depth, 'temp_'))) %>%
+    arrange(date, depth)
+}
+
+#' @title Get the surface, middle, and bottom depths for a given lake, 
+#' for plotting. Modified from Alison's code for picking surface, middle, 
+#' and bottom depths from observational data: 
+#' https://code.usgs.gov/wma/wp/lake-temperature-neural-networks/-/blob/master/3_assess/src/inspect_site.R#L81-93
+#' @description For a given lake, identify which depths represent the
+#' surface, middle, and bottom sections of the lake
+#' @param predictions_long A long-format dataset of GLM predictions at
+#' 0.5-meter intervals
+#' @return A named list of the surface, middle, and bottom depths to be used
+#' to plot data for a given lake.
+get_plotting_depths <- function(predictions_long) {
+  n_lambda <- 1
+  depth_scores <- predictions_long %>% group_by(depth) %>% tally() %>%
+    mutate(
+      n_score = n_lambda * n/max(n),
+      surface_score = n_score - abs(depth - 0),
+      middle_score = n_score - abs(depth - max(depth)/3),
```

Comment:
Curious as to why this is `/3` and not `/2` for the middle?

Reply:
That piece is from [Alison's original code]() -- my guess is that she was trying improve the odds of pulling a depth representative of the metalimnion/thermocline, rather than inadvertently ending up with another point that is representative of the hypolimnion (like the 'bottom' score). The metalimnion/thermocline, if present, is usually pretty narrow,  and the hypolimnion is a much larger percentage of the water column:
![image](https://user-images.githubusercontent.com/54007288/160892038-5e086175-fff2-4bca-a1f1-0cc80f95dc25.png)
[source](https://www.researchgate.net/figure/Cross-section-of-a-thermally-stratified-lake-showing-location-of-the-epilimnion-and_fig5_242740698)

So what I think it comes down to is that a point 1/3 of the way down the water column is more likely to fall into the metalimnion (or at least the uppermost hypolimnion) than a point 1/2 of the way down the water column

Reply:
Ahhh yes, excellent point! Thanks for the visual